src/app/checkout/navigateToOrderConfirmation.tsx

```diff
@@ -1,7 +1,18 @@
 import { noop } from 'lodash';
 
+interface RouteKeyMap<T = string> {
+    [key: string]: T;
+}
+
+const orderConfirmationRoutesMap: RouteKeyMap = {
+    '/embedded-checkout': '/embedded-checkout/order-confirmation',
+};
+
+const defaultOrderConfirmationRoute = '/checkout/order-confirmation';
+
 export default function navigateToOrderConfirmation(location = window.location): Promise<never> {
-    const url = `${location.pathname.replace(/\/$/, '')}/order-confirmation`;
+    const orderConfirmationRoute = Object.keys(orderConfirmationRoutesMap).filter(route => location.pathname.includes(route));
+    const url = orderConfirmationRoute.length > 0 ? orderConfirmationRoutesMap[orderConfirmationRoute[0]] : defaultOrderConfirmationRoute;
```

Comment:
We might not introduce `RouteKeyMap` and `orderConfirmationRoutesMap` as we currently only have two URLs for the order confirmation page. Once we have a new URL case, we can decide how we go from there.

`orderConfirmationRoutesMap[orderConfirmationRoute[0]]` seems not like a concrete solution.